spot_wrapper/wrapper.py

```diff
@@ -212,42 +313,124 @@ def _start_query(self):
-                         not self._spot_wrapper._last_trajectory_command_precise):
+                if (
+                    status
+                    == basic_command_pb2.SE2TrajectoryCommand.Feedback.STATUS_AT_GOAL
+                    or (
+                        status
+                        == basic_command_pb2.SE2TrajectoryCommand.Feedback.STATUS_NEAR_GOAL
+                        and not self._spot_wrapper._last_trajectory_command_precise
+                    )
+                ):
                     self._spot_wrapper._at_goal = True
                     # Clear the command once at the goal
                     self._spot_wrapper._last_trajectory_command = None
-                elif status == basic_command_pb2.SE2TrajectoryCommand.Feedback.STATUS_GOING_TO_GOAL:
+                elif (
+                    status
+                    == basic_command_pb2.SE2TrajectoryCommand.Feedback.STATUS_GOING_TO_GOAL
+                ):
                     is_moving = True
-                elif status == basic_command_pb2.SE2TrajectoryCommand.Feedback.STATUS_NEAR_GOAL:
+                elif (
+                    status
+                    == basic_command_pb2.SE2TrajectoryCommand.Feedback.STATUS_NEAR_GOAL
+                ):
                     is_moving = True
                     self._spot_wrapper._near_goal = True
+                elif (
+                    status
+                    == basic_command_pb2.SE2TrajectoryCommand.Feedback.STATUS_UNKNOWN
+                ):
+                    self._spot_wrapper._trajectory_status_unknown = True
+                    self._spot_wrapper._last_trajectory_command = None
                 else:
+                    self._logger.error(
+                        "Received trajectory command status outside of expected range, value is {}".format(
+                            status
+                        )
+                    )
                     self._spot_wrapper._last_trajectory_command = None
             except (ResponseError, RpcError) as e:
                 self._logger.error("Error when getting robot command feedback: %s", e)
                 self._spot_wrapper._last_trajectory_command = None
 
         self._spot_wrapper._is_moving = is_moving
 
+        # We must check if any command currently has a non-None value for its id. If we don't do this, this stand
+        # command can cause other commands to be interrupted before they get to start
+        if (
+            self._spot_wrapper.is_standing
+            and not self._spot_wrapper.is_moving
+            and self._spot_wrapper._last_trajectory_command is not None
+            and self._spot_wrapper._last_stand_command is not None
+            and self._spot_wrapper._last_velocity_command_time is not None
+            and self._spot_wrapper._last_docking_command is not None
+        ):
+            self._spot_wrapper.stand(False)
```

Comment:
I have to admit that its purpose is not entirely clear to me either. The whole AsyncIdle looks like it could do with a review to see if it's actually necessary.

Reply:
I'm just adding another parameter for it